pkg/vault/contracts/Vault.sol

```diff
@@ -1172,11 +1172,8 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
                 // The mint operation returns exactly `vaultWrappedDelta` shares. To do so, it withdraws underlying
                 // from the vault and returns the shares. So, the vault needs to approve the transfer of underlying
                 // tokens to the wrapper.
-                // Add convert error because mint can consume a different amount of tokens than we anticipated.
-                underlyingToken.forceApprove(
-                    address(wrappedToken),
-                    _addConvertError(amountInUnderlying + bufferUnderlyingSurplus)
-                );
+                // Add 2 because mint can consume up to 2 wei more tokens than we anticipated.
+                underlyingToken.forceApprove(address(wrappedToken), amountInUnderlying + bufferUnderlyingSurplus + 2);
```

Comment:
I've been thinking about this. I think we'll need to keep `forceApprove`, which prevents 2 issues:
- ABI incompatibility (return value)
- Token requiring spender allowance to be 0 when a value different than 0 is set

The second condition shouldn't have an issue with regular `approve`, since we're zeroing out the allowance below. But unfortunately the code will revert because of the first issue if we don't use this.

This way of doing it (approve 'exact' quantity needed + use allowance + zero out allowance) seems to be _very_ correct. I thought about a few optimizations:
- Avoid the sum (`amountInUnderlying + bufferUnderlyingSurplus + 2`); just give it a huge allowance since you're going to zero it out anyways afterwards. The problem with this is that the wrapper could take more underlying than what we were calculating, but that should revert since we measure how much it took, and if outside the requested limit the operation will revert right?
  - alt: make sum unchecked :face_with_spiral_eyes: this will very rarely overflow, and if it does we'll be granting less allowance than required, which will revert anyways
- Just leave the leftover allowance; in any case it should be small right? The problem here is that if the wrapper takes less tokens during the operation and the leftover allowance is significant, the wrapper can then drain those tokens without any sort of accounting.
- Minor, but here it goes: each `forceApprove` actually does 2 calls to the token: one with 0 allowance, and one with the desired allowance. Then, these two `forceApprove` actually are calling:
```
token.approve(wrapper, 0);
token.approve(wrapper,  amountInUnderlying + bufferUnderlyingSurplus + 2);
// mint
token.approve(wrapper, 0);
token.approve(wrapper,  0);
```

If we make a small modification to `SafeERC20` we can avoid the last call; we don't need to zero out the allowance twice.


Let's just think about this for now. No action needed in this PR.

Reply:
I tried replacing the sum by a big int value, but the code increased by 7 bytes, and there were no changes in terms of gas, so I'm leaving it as is for now.